src/Libraries/Microsoft.Extensions.Telemetry.Abstractions/buildTransitive/net8.0/Microsoft.Extensions.Telemetry.Abstractions.targets

```diff
@@ -1,2 +1,16 @@
 <Project>
+  <!-- This package should replace the Microsoft.Extensions.Logging.Abstractions source generator. -->
+  <Target Name="_Microsoft_Extensions_Logging_AbstractionsRemoveAnalyzers" 
+          Condition="'$(DisableMicrosoftExtensionsLoggingSourceGenerator)' == 'true'"
+          AfterTargets="ResolveReferences">
+    <ItemGroup>
+      <_Microsoft_Extensions_Logging_AbstractionsAnalyzer Include="@(Analyzer)" Condition="'%(Analyzer.NuGetPackageId)' == 'Microsoft.Extensions.Logging.Abstractions' Or
+                                                                                           ('%(Analyzer.AssemblyName)' == 'Microsoft.Extensions.Logging.Generators' and '%(Analyzer.NuGetPackageId)' == 'Microsoft.AspNetCore.App.Ref')" />
```

Comment:
Isn't it enough to check the assembly's name only? Why do we need to check the package Id at all?

Reply:
It's mostly a conservative approach. Without the condition, this becomes a big hammer for any generator with that name in the future, even if it is a custom logging generator with the same name shipped via a 3-rd party package. By specifying the packageId's we are allowed to remove, we are effectively limiting the effect to only do it when we really mean to remove it, which is either when it's replacing the generator from the logging abstractions package or the one from the targeting pack.

I'm fine to remove the condition if people really feel strongly about it, but here I was mostly just following precedence which I believe is the correct one as well as it limits the impact.

Reply:
> I don't think we should care much if anyone in the future introduced a source gen with the same assembly's name. I am concerned more about the other scenario, if anyone in the future creates some package similar to AspNetCore one and then they start see the problem.

Fair enough, I can remove the condition just in case.'

> Another point I want to discuss is, do we need to have a property that does the reverse? If anyone wants to use this package but use the runtime generator for any reason.

This sounds reasonable to me but let's discuss and do it in a separate PR if that's okay.

Reply:
Just checking AssemblyName ended up not being enough because analyzers resolved via nuget package restore doesn't have that metadata set, so my condition for selecting the analyzer to remove ended up matching either the AssemblyName (for the cases the analyzer resolved was coming from the targeting pack), or the packageId (for the cases the analyzer was resolved from nuget package restore)